### PR TITLE
Add pipenv support

### PR DESCRIPTION
**Pull request by wusyong**
These files are for better package version management and environment
setup in the future. It can even integrate with DOCKERFILE to deploy
server automatically.

This change will not affect original workflow. Install dependencies
with `requirements.txt` is still available.

### CODE REVIEW

File: Pipfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+[[source]]
+name = "pypi"
+url = "https://pypi.org/simple"
```
Comment (jserv) [inline Pipfile]: Why `simple` here?

**Comment by jserv** [comment]
I have no idea about file `Pipfile.lock`. @ender503, can you revise it?

**Comment by ender503** [comment]
When using the `Pipfile` to install the package dependencies, it will generate the `Pipfile.lock` file that contains the each package versions, commit `Pipfile.lock` file  to ensure everyone is using the same package version.

**Comment by ender503** [comment]
`Pipfile` is another way to install the package dependencies, like `requirements.txt`.
@wusyong , could you add the guide to install the package dependencies with the `Pipfile` in the `README.md` ? 😃

File: Pipfile (hunk 1 above)
Comment (wusyong) [reply Pipfile]: It's default config for pipenv. And other source url are usually the same.

**Comment by jserv** [comment]
I defer to @ender503 about the adoption of file `Pipfile.lock`.

**Pull request merged**